src/cc_catalog_airflow/dags/provider_api_scripts/iconfinder.py

```diff
@@ -0,0 +1,199 @@
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 100
+DELAY = 5.0
+RETRIES = 3
+PROVIDER = "iconfinder"
+SEARCH_ENDPOINT = "https://api.iconfinder.com/v4/icons/search"
+ITEM_ENDPOINT = "https://api.iconfinder.com/v4/icons/"
+API_KEY = os.getenv("ICONFINDER_API_KEY")
+
+delay_request = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+DEFAULT_QUERY_PARAM = {
+    "query": ":",
+    "count": LIMIT,
+    "offset": 0,
+    "premium": 0
+}
+
+HEADERS = {
+    'authorization': f'Bearer {API_KEY}'
+}
+
+
+def main():
+    offset = 0
+    condition = True
+    while condition:
+        query_param = _get_query_param(
+            offset=offset
+        )
+        response_json = _request_handler(
+            params=query_param
+        )
+        icon_batch = response_json.get("icons")
+        if type(icon_batch) == list and len(icon_batch) > 0:
+            _process_icon_batch(icon_batch)
+            logger.info(
+                f"Images collected in this batch: {image_store.total_images}"
+            )
+            offset += LIMIT
+        else:
+            condition = False
+    image_store.commit()
+    logger.info(f"total images : {image_store.total_images}")
+
+
+def _get_query_param(
+        offset=0,
+        default_query_param=DEFAULT_QUERY_PARAM,
+        ):
+    query_param = default_query_param.copy()
+    query_param["offset"] = offset
+    return query_param
+
+
+def _request_handler(
+        endpoint=SEARCH_ENDPOINT,
+        params=None,
+        headers=HEADERS,
+        retries=RETRIES
+        ):
+    for retry in range(retries):
+        response = delay_request.get(
+            endpoint,
+            params=params,
+            headers=headers
+        )
+        if response.status_code == 200:
+            try:
+                response_json = response.json()
+                break
+            except Exception as e:
+                logger.error(f"Request failed due to {e}")
+                response_json = None
+        else:
+            response_json = None
+
+    return response_json
+
+
+def _process_icon_batch(icon_batch):
+    for icon in icon_batch:
+
+        icon_id = icon.get("icon_id")
+        if icon_id is None:
+            continue
+
+        icon_data = _request_handler(
+            endpoint=ITEM_ENDPOINT+str(icon_id)
+        )
+        if icon_data is None:
+            continue
+
+        iconset = icon_data.get("iconset")
+        if iconset is None:
+            continue
+        license_url = _get_license(
+            iconset.get("license")
+        )
+        if license_url is None:
+            continue
+        foreign_landing_url = (
+            "https://www.iconfinder.com/icons/" + str(icon_id)
+        )
+        image_url, thumbnail_url, height, width = _get_images(
+            icon_data.get("raster_sizes")
+        )
+        if image_url is None:
+            continue
```

Comment:
I'd rather keep these 'skips' restricted to the `ImageStore` if possible.  This is because it might be that we later change what's required to keep an image, and what isn't, and it's easier to make such a transition if all the knowledge about what is necessary to successfully store an image is restricted to one part of the codebase.